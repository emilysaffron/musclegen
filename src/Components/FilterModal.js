import { useState } from "react";

import getExerciseTargets from "../Helpers/getExerciseTargets";
const FilterModal = (filterExercises, half) => {
  const [filterState, toggleFilterState] = useState(false);

  return filterState ? (
    <div>
      {getExerciseTargets(filterExercises, half)}
      <button onClick={() => toggleFilterState(!filterState)} />
    </div>
  ) : (
    <button onClick={() => toggleFilterState(!filterState)} />
  );
};
export default FilterModal;
