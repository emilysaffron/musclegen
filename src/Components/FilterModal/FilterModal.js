import { useState } from "react";
import styled from "@emotion/styled";

import getExerciseTargets from "../../Helpers/getExerciseTargets";

const StyledModal = styled.div`
  background: #cea8bb;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  color: black;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: max-content;
  border-radius: 20px;
  position: -webkit-sticky;
  position: sticky;
  top: 1rem;
  @media (max-width: 400px) {
    z-index: 100;
    width: 80%;
    margin-left: 50px;
  }
`;

const OpenButton = styled.button`
  background: #cea8bb;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  color: black;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: max-content;
  border: 2px solid black;
  position: -webkit-sticky;
  position: sticky;
  top: 1rem;
  border-radius: 20px;
  cursor: pointer;
  @media (max-width: 400px) {
    z-index: 100;
    width: 100%;
    margin-left: 50px;
  }
`;

const ClearFilter = styled.button`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 5px;
  margin: 5px;
  align-items: center;
  border-radius: 20px;
  width: max-content;
  cursor: pointer;
`;

const FilterModal = ({ filterExercises, half }) => {
  const [filterState, toggleFilterState] = useState(false);

  return filterState ? (
    <StyledModal
      key={Math.random()}
      onClick={() => toggleFilterState(!filterState)}
    >
      {getExerciseTargets(filterExercises, half)}
      <ClearFilter key={Math.random()} onClick={() => filterExercises("")}>
        Clear Filter
      </ClearFilter>
    </StyledModal>
  ) : (
    <OpenButton onClick={() => toggleFilterState(!filterState)}>
      Filter Exercises
    </OpenButton>
  );
};
export default FilterModal;
