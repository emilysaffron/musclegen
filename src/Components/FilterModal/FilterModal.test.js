import renderer from "react-test-renderer";
import FilterModal from "./FilterModal";

it("FilterModal Snapshot", () => {
  const tree = renderer
    .create(<FilterModal filterExercises="filterExercises" half="half" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
