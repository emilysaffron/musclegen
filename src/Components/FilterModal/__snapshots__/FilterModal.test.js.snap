// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterModal Snapshot 1`] = `
<button
  className="css-15i479a"
  onClick={[Function]}
>
  Filter Exercises
</button>
`;
